### DISCUSSION TITLE
Optional connect?

### DISCUSSION THREAD

**Discussion by alucardu** [prisma/prisma#18161] [Q&A] [answered]
I have a relation between my `user` and `session` model. But the relation is optional. I want to be able to create a user that's not connected to a session.

```gql
export const userResolvers = {
  Mutation: {
    createUser: async (_, args) => {
      try {
        if (args.sessionId) {
          return await prisma.user.create({

            data: {
              name: args.name,
              session: {
                connect: { id: Number(args.sessionId) },
              }
            }
          })
        } else {
          return await prisma.user.create({

            data: {
              name: args.name,
            }
          })
        }
      } catch (err) {
        throw new Error(`Username ${args.name} already exists`);
      }
    },
  },
};
```
This works, but the syntax is ugly. I don't think create or connect is a good option since I don't want to create a session. Is there a better way to write this syntax?

**Comment by alucardu** [comment, marked as answer]
I ended up with a different solution. I create a user, and a session (without any relation to each other). When both mutations are resolved I call a `connectSessionAndHost` which takes the user and session I just created and connect a user to the session host:

```gql
connectSessionHost: async (_, args) => {
  return await prisma.session.update({
    where: {
      id: Number(args.sessionId)
    },
    data: {
      host: {
        connect: {
          id: Number(args.userId)
        }
      }
    }
  })
},
```
I don't know if there's a better/different way to do this, but this does what I wanted.

**Reply by nurul3101** [reply]
Can you share the user and session schema as well? If the relation is optional, then it shouldn't be required to use connect while creating a user.
